packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1717,17 +1717,19 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   @override
   void didPopNext(Route<dynamic> nextRoute) {
     super.didPopNext(nextRoute);
-    changedInternalState();
+    changedInternalState(false);
   }
 
   @override
-  void changedInternalState() {
+  void changedInternalState([bool markNeedsBuild = true]) {
```

Comment:
I'm not 100% sure if this is the way we should organize this, but I'm interested to see if it fixes the problem without breaking any tests. If so we can worry about cleaning up the architecture.

Reply:
I played around and looked at the tests, but seems this would be a dead end. I agree this is ugly and definitely not the best approach.

I haven't really spent a lot of time looking at the engine and frameworks underlying code, so I'm unsure how to go about finding a better solution.

Maybe someone with more knowledge on the workings of routes, can gleam why my latest changes work and come up with a better solution.

Reply:
Also, my latest changes fixed tests and does work, but I'm not exactly sure if we even want this merged, thus I haven't added a test yet (Also not sure where would be appropriate to add the test).

Reply:
This seems weird for 2 reasons.
1, the caller of `changedInternalState` should not know about markneedsbuild because it is a implementation detail in ModalRoute
2, rebuild should not crash the app.

Do you know what is exactly causing the issue?

Reply:
As mentioned in response to Justin, I am not knowledgeable about the engine and framework.

I was hoping you'd pick it up and I could close this one.